test/runtests.jl

```diff
@@ -226,14 +226,25 @@ end
         end
 
         @testset "give an error if the struct contains internal constructors 4" begin
-            @test_throws internal_construcor_error begin
+            @test_throws internal_constructor_error begin
                 @macroexpand @auto_hash_equals_cached struct T156
                     function T156() where {T}
                         new()
                     end
                 end
             end
         end
+
+        @testset "check compatibility with default constructor" begin
+            @auto_hash_equals_cached struct S268
+                x::Int
+            end
+            @test S268(2.0).x === 2
+            @auto_hash_equals_cached struct S269{T <: Any}
+                x::T
+            end
+            @test S269{Int}(2.0).x === 2
```

Comment:
i  think this is already covered elsewhere, but you might combine this test with testing the other situation too, to make sure both behaviors are preserved:
```suggestion
            @test S269{Int}(2.0).x === 2
            @test S269(2.0).x === 2.0
```